modules/tinymce/src/core/test/ts/browser/delete/BlockMergeBoundaryTest.ts

```diff
@@ -72,6 +72,12 @@ describe('browser.tinymce.core.delete.BlockMergeBoundary', () => {
       const boundaryOpt = readBlockBoundary(true, [ 1, 0 ], 1);
       assertBlockBoundaryNone(boundaryOpt);
     });
+
+    it('TINY-10590: If the cursor is between an internal block and a text-node it should not count as a block boundary', () => {
```

Comment:
Flush this out with more tests. Like backspace on: `<div>A<p>|B</p>C</div>` or delete on `<div>A<p>B|</p>C</div>` so the bug is more wide that just this case. So think the `isNotAncestorial` will need to be expanded a bit to handle the other cases here.

Reply:
I think you could use `Compare.contains` here so if from contains to or to contains from then it should not do anything.